test/node/connection-event.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const test = require('tap');
+
+const jstp = require('../..');
+
+const name = 'testApp';
+
+const interfaces = {
+  iface: {}
+};
+
+const app = {
+  name,
+  interfaces
+};
+
+const application = new jstp.Application(app.name, app.interfaces);
+let server;
+let client;
+
+test.beforeEach((done) => {
+  server = jstp.tcp.createServer(0, [application], app.authCallback);
+  server.listen(() => {
+    const port = server.address().port;
+    client = jstp.tcp.createClient({ host: 'localhost', port });
+    done();
+  });
+});
+
+test.afterEach((done) => {
+  client.disconnect();
+  server.close();
+  done();
+});
+
+const iface = 'iface';
+const eventName = 'someEvent';
+const args = ['firstArgument', 'secondArgument'];
+
+test.test('server must process an event', (test) => {
+  client.connectAndHandshake(app.name, null, null, (error, connection) => {
+    test.assertNot(error, 'must connect to server');
+
+    server.getClients()[0].on('event', (event) => {
+      test.strictEqual(event.remoteEventName, eventName,
```

Comment:
Here we only compare name and args, but current implementations sends more [data](https://github.com/metarhia/jstp/blob/master/lib/connection.js#L534). I not sure if we need to check all of the data to be correct as a lot of data there is redundant (but that's a matter of different issue) but this may be the case as we will still probably need to change this test if we change events. @metarhia/jstp-core what do you think?